src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentChildContentDiagnosticPass.cs

```diff
@@ -52,26 +49,36 @@ public override void VisitComponentChildContent(ComponentChildContentIntermediat
             // because the parameter name can be implicit, and it doesn't work well when nested.
             if (node.IsParameterized)
             {
-                for (var i = 0; i < Ancestors.Count - 1; i++)
+                var ancestors = Ancestors;
+
+                var parentComponent = (ComponentIntermediateNode)ancestors[0];
+                ancestors = ancestors[1..];
+
+                while (!ancestors.IsEmpty)
                 {
-                    var ancestor = Ancestors[i] as ComponentChildContentIntermediateNode;
-                    if (ancestor != null &&
-                        ancestor.IsParameterized &&
-                        string.Equals(node.ParameterName, ancestor.ParameterName, StringComparison.Ordinal))
+                    if (ancestors is
+                        [
+                            ComponentChildContentIntermediateNode { IsParameterized: true } ancestor,
```

Comment:
>IsParameterized: true [](http://example.com/codeflow?start=68&length=21)

could the ParameterName check be moved in here too?

Reply:
Nope. Pattern matching doesn't match against variables.